Make failure caused by usage of "O_CREAT" flag on Ubuntu 16.04

@alistairking discovered this make failure during make on Ubuntu 16.04

```
In file included from /usr/include/fcntl.h:289:0,
                 from bs_transport_cache.c:32:
In function 'open',
    inlined from 'bs_transport_cache_create' at bs_transport_cache.c:180:13:
/usr/include/x86_64-linux-gnu/bits/fcntl2.h:50:4: error: call to '__open_missing_mode' declared with attribute error: open with O_CREAT or O_TMPFILE in second argument needs 3 arguments
    __open_missing_mode ();
```

Potential fix discussion found at: https://github.com/pantheon-systems/fusedav/issues/204

--- a/lib/transports/bs_transport_cache.c
+++ b/lib/transports/bs_transport_cache.c
@@ -177,7 +177,7 @@ int bs_transport_cache_create(bgpstream_transport_t *transport)
     // local cache file doesn't exist
 
     // try to create a lock file for cache writing
-    lock_fd = open(STATE->lock_file_path,O_CREAT|O_EXCL);
+    lock_fd = open(STATE->lock_file_path,O_CREAT|O_EXCL, 0644);
 
     if(lock_fd<0){
       // lock file creation failed: other thread is still writing the cache

